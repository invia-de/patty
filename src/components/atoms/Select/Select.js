import React from 'react';
import PropTypes from 'prop-types';

import cx from '../../../utils/classnames';
import { ArrowDown } from '../Icon/Icon';

import styles from './select.module.scss';

/**
 * @author [Roman Semko](mailto:roman.semko-extern@invia.de)
 */
export default function Select({ children, className, ...props }) {
  return (
    <div className={cx(styles.select, className)}>
      <ArrowDown className={styles.icon} />
      <select {...props}>{children}</select>
    </div>
  );
}

Select.propTypes = {
  /** additional classNames you want to add */
  className: PropTypes.string,
  /** required due to accessibility */
  children: PropTypes.node.isRequired
};
