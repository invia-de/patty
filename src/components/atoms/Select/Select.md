```jsx
<Select value="b">
  <option value="a">Alfa</option>
  <option value="b">Beta</option>
  <option value="c">Gamma</option>
  <option value="d">Delta</option>
</Select>
```
