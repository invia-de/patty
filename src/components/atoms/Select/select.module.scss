@import 'theme.scss';

$select-background: $white !default;
$select-border-color: #afc7f3 !default;
$select-icon-color: $aqua !default;

.select {
  position: relative;
  display: inline-block;
  background-color: $select-background;
  select {
    position: relative;
    z-index: 100;
    width: 196px;
    //   height: 16px;
    padding: 7px 27px 7px 9px;
    color: $blue;
    font-size: 14px;
    line-height: 16px;
    font-weight: bold;
    border: 1px solid $select-border-color;
    background: transparent;
  }
}

.icon {
  position: absolute;
  z-index: 50;
  right: 9px;
  top: 10px;
  fill: $select-icon-color;
}
