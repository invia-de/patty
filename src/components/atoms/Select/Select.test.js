import React from 'react';
import Select from './Select';
import { render } from '@testing-library/react';

test('Select renders', () => {
  const { container } = render(<Select />);
  expect(container.firstChild).toMatchSnapshot();
});
