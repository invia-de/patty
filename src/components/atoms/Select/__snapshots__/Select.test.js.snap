// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select renders 1`] = `
<div
  class="select"
>
  <svg
    aria-hidden="true"
    class="icon icon"
    viewBox="0 0 512 512"
  >
    <path
      d="M430.394 88.254l-175.45 175.403L80.549 86.803l-80.001 80 259.03 259.056 252.968-253.016-82.152-84.589z"
    />
  </svg>
  <select />
</div>
`;
